### PR TITLE
Loki: Fixes chained template variable queries for scene apps

### PR DESCRIPTION
**Pull request by torkelo**
ScopedVars is required for variable interpolation to work when used from scenes library.

Will open a backward compatible solution for the same problem for external data sources.

* Adds the missing type to metricFindQuery
* Adds scopedVars to LegacyMetricFindQueryOptions
* Updates loki variable query to pass scopedVars

### CODE REVIEW

File: public/app/features/variables/utils.ts [hunk 1]
```diff
@@ -131,8 +131,14 @@ export function getTemplatedRegex(variable: QueryVariableModel, templateSrv = ge
   return templateSrv.replace(variable.regex, {}, 'regex');
 }
 
-export function getLegacyQueryOptions(variable: QueryVariableModel, searchFilter?: string, timeSrv = getTimeSrv()) {
-  const queryOptions: any = { range: undefined, variable, searchFilter };
+export function getLegacyQueryOptions(
+  variable: QueryVariableModel,
+  searchFilter: string | undefined,
+  timeSrv: TimeSrv,
```
Comment (dprokop) [inline public/app/features/variables/utils.ts:137]: You are removing default value here, think this may be problematic.

**Review by dprokop — APPROVED** [review]

File: public/app/features/variables/utils.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/variables/utils.ts:137]: @dprokop it was only used from one location where it was always passed

**Comment by mckn** [comment]
This won't break any plugins during runtime? It, might, only be when they update the grafana-package dependencies. Right? Is there any need to add anything to the migration guide?

**Review by matyax — COMMENTED** [review]
From the Logs point of view this looks great! 👍

**Comment by torkelo** [comment]
> This won't break any plugins during runtime? It, might, only be when they update the grafana-package dependencies. Right? Is there any need to add anything to the migration guide?

@mckn no, no runtime change here. Only a minor type fix. Should not even cause any type errors (unless a plugin has manually created a type for the metricFindQuery options argument)

**Pull request merged**